kpatch-build/create-klp-module.c

```diff
@@ -208,8 +239,23 @@ static void create_klp_relasecs_and_syms(struct kpatch_elf *kelf, struct section
 		ALLOC_LINK(rela, &klp_relasec->relas);
 		rela->sym = sym;
 		rela->type = krelas[index].type;
-		rela->offset = krelas[index].offset;
-		rela->addend = krelas[index].addend;
+		if (!strcmp(base->name, ".toc"))
+			rela->offset = toc_offset;
+		else
+			rela->offset = krelas[index].offset;
+
+		/*
+		 * gcc6 adds offset for 0x8 for every local function entry in
+		 * .toc section, for avoiding setup of toc but when the previously
+		 * local function becomes global, toc is anyways setup. So remove
+		 * the wrong addend.
+		 */
+		if (!strcmp(base->name, ".toc") &&
+			rela->sym->type == STT_FUNC && rela->sym->bind == STB_LOCAL) {
```

Comment:
Here ```rela->addend``` check for the ```klp.rela.toc``` relocation entries, where the previously local functions are global now, but their ``` sym->bind == STB_LOCAL ``` and the ```sym.st_shndx == 0xff20 ``` a.k.a not equal to ``` sym.st_shndx != SHN_UNDEF```. The latter check for ```sym.st_shndx``` will return false positive and also given that they are not part of current livepatch module the instructions between global and local entry point of these function can not be decoded.